pkg/validations/extendedversion.go

```diff
@@ -2,21 +2,41 @@ package validations
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
 
 	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
 	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/signature"
 	"github.com/aws/eks-anywhere/release/api/v1alpha1"
 )
 
 // ValidateExtendedK8sVersionSupport validates all the validations needed for the support of extended kubernetes support.
-func ValidateExtendedK8sVersionSupport(_ context.Context, _ *anywherev1.Cluster, bundle *v1alpha1.Bundles, _ kubernetes.Client) error {
+func ValidateExtendedK8sVersionSupport(_ context.Context, clusterSpec *anywherev1.Cluster, bundle *v1alpha1.Bundles, _ kubernetes.Client) error {
```

Comment:
looks like the `clusterSpec` is needed but only for 2 fields within it. I don't recommend taking in the whole cluster spec as a dependency.

Reply:
As discussed offline, I am not passing `clusterSpec` as a pointer.